tests/samplers_tests/test_nsga2.py

```diff
@@ -535,15 +537,104 @@ def objective(trial: optuna.Trial) -> float:
     assert len(study.trials) == n_trials
 
 
-@pytest.mark.parametrize("crossover_name", ["undx", "spx"])
-def test_crossover_invalid_population(crossover_name: str) -> None:
+@pytest.mark.parametrize("crossover", [UNDXCrossover(), SPXCrossover()])
+def test_crossover_invalid_population(crossover: BaseCrossover) -> None:
     n_objectives = 2
     n_trials = 8
 
     with pytest.raises(ValueError):
-        sampler = NSGAIISampler(population_size=2, crossover=crossover_name)
+        sampler = NSGAIISampler(population_size=2, crossover=crossover)
         study = optuna.create_study(directions=["minimize"] * n_objectives, sampler=sampler)
         study.optimize(
             lambda t: [t.suggest_float(f"x{i}", 0, 1) for i in range(n_objectives)],
             n_trials=n_trials,
         )
+
+
+@pytest.mark.parametrize(
+    "crossover",
+    [
+        UniformCrossover(),
+        BLXAlphaCrossover(),
+        SPXCrossover(),
+        SBXCrossover(),
+        VSBXCrossover(),
+        UNDXCrossover(),
+    ],
+)
+def test_crossover_numerical_distribution(crossover: BaseCrossover) -> None:
+
+    study = optuna.study.create_study()
+    rng = np.random.RandomState()
+    search_space = {"x": FloatDistribution(1, 10), "y": IntDistribution(1, 10)}
+    parent_params = np.array([[1.0, 2], [3.0, 4]])
+
+    if crossover.n_parents == 3:
+        parent_params = np.append(parent_params, [[5.0, 6]], axis=0)
+
+    child_params = crossover.crossover(parent_params, rng, study, search_space)
+    assert child_params.ndim == 1
+    assert len(child_params) == len(search_space)
+    assert np.nan not in child_params
+    assert np.inf not in child_params
+
+
+def test_crossover_categorical_distribution() -> None:
+
+    study = optuna.study.create_study()
+    search_space: Dict[str, BaseDistribution] = {
+        "x": CategoricalDistribution(choices=["a", "c"]),
+        "y": CategoricalDistribution(choices=["b", "d"]),
+    }
+    parent_params = np.array([["a", "b"], ["c", "d"]])
+
+    crossover = UniformCrossover()
+    child_params = crossover.crossover(parent_params, np.random.RandomState(), study, search_space)
+
+    assert child_params.ndim == 1
+    assert len(child_params) == len(search_space)
+    assert all([isinstance(param, str) for param in child_params])
+
+    # Is child param from correct distribution?
+    search_space["x"].to_internal_repr(child_params[0])
+    search_space["y"].to_internal_repr(child_params[1])
```

Comment:
Could you please explain what this part is testing?

Reply:
Essentially, we are checking if the position of parameters was not switched in child and thus if masking for uniform crossover works correctly. For example - if one of the choices possible for `y` param: `[b, d]` would end up on 0 index in child array, then `to_internal_repr` of `x`'s distribution would rise an error.